python/seldon_core/microservice.py

```diff
@@ -188,6 +195,11 @@ def setup_logger(log_level: str) -> logging.Logger:
     flask_logger = logging.getLogger("werkzeug")
     flask_logger.setLevel(log_level_num)
 
+    if getenv_as_bool(FILTER_METRICS_ACCESS_LOGS_ENV_NAME, default=False):
+        flask_logger.addFilter(MetricsEndpointFilter())
+        gunicorn_logger = logging.getLogger("gunicorn.access")
```

Comment:
Is there any chance this fails if Gunicorn is not being used (e.g. as in returning `None`)?

Reply:
I tested both :)

This is anyway called before the decision for flask/gunicorn is made.

Reply:
Both: with and without `--debug` flag